### PR TITLE
[MX-396] [MX-364] Create 'Payment' menu in profile tab

### PR DESCRIPTION
**Pull request by ds300**
https://artsyproduct.atlassian.net/browse/MX-364

This PR is getting too big so I want to merge a few of these changes before doing follow up.

## Changes in this PR

- New Payments menu item in the profile tab.
- New 'Add new card' screen. (This uses easy-peasy to manage the form state. I originally did it only as an experiment and was planning to convert to Formik but honestly with the computed properties and actions I prefer it to Formik so i left it as-is)
- Create component for adding the new style generic page header `PageWithSimpleHeader`
- Create new modal component `FancyModal` based on the core react-native `Modal` for doing the fancypants iOS 13 presentation style (should work on android too)
- Create new `Select` component. This is part of palette but I haven't put it in the palette repo because I'm not convinced that doing that provides any value while we only have one react-native app. Happy for someone else to do that as follow-up if they think it's worthwhile.
- Fixed a minor UI bug with the home hero
- Applied the UI fixes from @samchieng's QA the other day
- Fixed the `Input` component so that the `KeyboardAvoidingView` can keep the whole thing in view (+ padding!)

## Planned Follow-up

- [ ] Add an optional search field to the `Select` component
- [ ] Tests!
- [ ] Look into proving useful error messages to the user on the credit card form
- [ ] Give the `Select` a `ref` prop so we can open the country select programatically after the user hits enter on the 'State, province, etc' field. Also make sure focus/tabbing works
- [x] ~Look into updating `tipsi-stripe` in the hopes that they've made the credit card input behave more like a text input (we could use an onEnter callback, and it has weird focus/blur behaviour too)~
- [ ] Placeholder for payment screen

### FancyModal

Here it is being used on the collections page with a custom sheet height

new-collections-modal

### Select

This also demonstrates `FancyModal` with the default sheet height (full screen)

new-select

### Payment

image
image

### New Card

new-card-form

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Missing Test Files:

- `src/lib/Components/__tests__/CreditCardDetails-tests.tsx`
- `src/lib/Components/__tests__/FancyModal-tests.tsx`
- `src/lib/Components/__tests__/PageWithSimpleHeader-tests.tsx`
- `src/lib/Components/__tests__/Select-tests.tsx`
- `src/lib/Scenes/MyProfile/Components/__tests__/MyProfileMenuItem-tests.tsx`
- `src/lib/Scenes/MyProfile/__tests__/MyProfilePayment-tests.tsx`
- `src/lib/Scenes/MyProfile/__tests__/MyProfilePaymentNewCreditCard-tests.tsx`
- `src/lib/utils/__tests__/useInterval-tests.ts`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Missing Test Files:

- `src/lib/Components/__tests__/CreditCardDetails-tests.tsx`
- `src/lib/Components/__tests__/FancyModal-tests.tsx`
- `src/lib/Components/__tests__/PageWithSimpleHeader-tests.tsx`
- `src/lib/Components/__tests__/Select-tests.tsx`
- `src/lib/Scenes/MyProfile/Components/__tests__/MyProfileMenuItem-tests.tsx`
- `src/lib/Scenes/MyProfile/__tests__/MyProfilePayment-tests.tsx`
- `src/lib/Scenes/MyProfile/__tests__/MyProfilePaymentNewCreditCard-tests.tsx`
- `src/lib/utils/__tests__/useInterval-tests.ts`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/MyProfile/MyProfilePaymentNewCreditCard.tsx [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+        input: {
+          oneTimeUse: false,
+          token,
+        },
+      },
+    })
+  })
+}
+
+const COUNTRY_SELECT_OPTIONS = [
```
Comment (ashfurrow) [inline src/lib/Scenes/MyProfile/MyProfilePaymentNewCreditCard.tsx:273]: I'm surprised that we don't already have this in the app. It looks like [the existing form for Auctions bid flow](https://github.com/artsy/eigen/blob/9bd6b54116f2586a9053317cca5ff7ea2229d816/src/lib/Components/Bidding/Screens/SelectCountry.tsx) uses a search-as-you-type API call to Google – huh! I like this quite a bit better.

File: src/lib/Scenes/MyProfile/MyProfilePaymentNewCreditCard.tsx [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+  { label: "Guadeloupe", value: "GP" },
+  { label: "Guam", value: "GU" },
+  { label: "Guatemala", value: "GT" },
+  { label: "Guernsey", value: "GG" },
+  { label: "Guinea", value: "GN" },
+  { label: "Guinea-Bissau", value: "GW" },
+  { label: "Guyana", value: "GY" },
+  { label: "Haiti", value: "HT" },
+  { label: "Heard Island and Mcdonald Islands", value: "HM" },
+  { label: "Holy See (Vatican City State)", value: "VA" },
```
Comment (ashfurrow) [inline src/lib/Scenes/MyProfile/MyProfilePaymentNewCreditCard.tsx:369]: I guess the nice thing about the API call for country select is that it's future-compatible if they move the Holy See back to Avignon 🤣

File: src/lib/Components/FilterModal.tsx [hunk 3]
```diff
@@ -55,54 +55,45 @@ export const FilterModalNavigator: React.SFC<FilterModalProps> = props => {
...
-                    {getApplyButtonCount()}
-                  </ApplyButton>
-                </ApplyButtonContainer>
-              </ModalInnerView>
-            </>
-          </TouchableWithoutFeedback>
-        </Modal>
-      )}
-    </>
+    <FancyModal visible={isFilterArtworksModalVisible} onBackgroundPressed={handleClosingModal} maxHeight={550}>
```
Comment (ashfurrow) [inline src/lib/Components/FilterModal.tsx:58]: /cc @brainbicycle these are the only changes you'd need to merge in for #3507.

File: src/lib/Components/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                  </PopIn>
+                ) : null}
+              </Flex>
+            </TouchableHighlight>
+          )}
+        ></FlatList>
+      </FancyModal>
+    </>
+  )
+}
```
[the remark this answers is not in the document]
Comment (ds300) [reply src/lib/Components/Select.tsx:142]: added to knowledge share ideas!

**Comment by ds300** [comment]
Thanks for the helpful feedback 🙌 

Yea it works fine on non-rounded-corner devices. Good thinking on the `KeyboardAvoidingView` though! I hadn't considered that, and indeed it didn't work properly with text inputs in the modal. I made some adjustments this morning to add basic support for keyboard avoiding in full-height fancy modals, but if we want to support non-full-height fancy modals it would require some native work.

**Comment by MounirDhahri** [comment]
very nice PR 🎉

**Pull request merged**